src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java

```diff
@@ -185,4 +199,43 @@ public void validateAuthorization(final EventType original, final EventTypeBase
         validateAuthorization(newAuth);
     }
 
+    public AdminAuthorization getAdmins() {
+        final List<Permission> allPermissions = authorizationDbRepository.listAdmins();
+        final List<AuthorizationAttribute> adminPermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.ADMIN)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> readPermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.READ)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> writePermissions = allPermissions.stream()
+                .filter(p -> p.getOperation() == AuthorizationService.Operation.WRITE)
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        return new AdminAuthorization(adminPermissions, readPermissions, writePermissions);
+    }
+
+    public void updateAdmins(final AdminAuthorization newAdmins) throws InsufficientAuthorizationException {
+        if (newAdmins.getAdmins().isEmpty() || newAdmins.getWriters().isEmpty() || newAdmins.getReaders().isEmpty()) {
```

Comment:
Isn't it checked on controller level by spring? Also you pass view objects to service level.

Reply:
7933617
Still an issue with spring not checking the minimum size as I expected, I'm on it

Reply:
62eb15b